src/point_one/fusion_engine/parsers/fusion_engine_framer.cc

```diff
@@ -189,14 +200,23 @@ p1_ssize_t FusionEngineFramer::OnByte(bool quiet) {
       // Compute the full message size. If the message is too large to fit in
       // the buffer, we cannot parse it. Otherwise, start collecting the
       // message payload.
+      //
+      // Note that while we compute the full message size here, we intentionally
+      // do the "too big" check below with the payload size. On a 32b machine,
```

Comment:
I don't understand this comment, it seems to explain exactly the case for not doing the new logic only...?

Reply:
The message size is defined as payload + header, there's no reason for it to ever be set differently. Checking the condition this way using the payload size is guaranteed to include 100% of possible cases, so also doing a check on the message size variable would just be redundant.